api/types/provisioning.go

```diff
@@ -62,21 +62,39 @@ type ProvisionToken interface {
 	// GetSuggestedLabels returns the set of labels that the resource should add when adding itself to the cluster
 	GetSuggestedLabels() Labels
 
-	// V1 returns V1 version of the resource
-	V1() *ProvisionTokenV1
 	// String returns user friendly representation of the resource
 	String() string
+
+	// V3 returns the V3 representation of a ProvisionToken
+	// This exists for the transition period where V2 tokens can still be
```

Comment:
>This exists for the transition period
So both `V3()` and `V2()` are for the transition period, or the documentation is wrong?

Reply:
If a V2 exists in the backend and a client asks for a list of Tokens will the returned type be a V3 or V2? 
If we convert and send it to the client as V3 we might have some troubles if he wants to update the token spec because as far as he knows the type in the backend is v3.